spec/requests/casa_admins_spec.rb

```diff
@@ -0,0 +1,88 @@
+require "rails_helper"
+
+RSpec.describe "/casa_admins", type: :request do
+  describe "GET /casa_admins/:id/edit" do
+    context "logged in as admin user" do
+      it "can successfully access a casa admin edit page" do
+        sign_in_as_admin
+
+        get edit_casa_admin_path(create(:casa_admin))
+
+        expect(response).to be_successful
+      end
+    end
+
+    context "logged in as a non-admin user" do
+      it "cannot access a casa admin edit page" do
+        sign_in_as_volunteer
+
+        get edit_casa_admin_path(create(:casa_admin))
+
+        expect(response).to redirect_to root_path
+        expect(response.request.flash[:notice]).to eq "Sorry, you are not authorized to perform this action."
+      end
+    end
+
+    context "unauthenticated request" do
+      it "can successfully access a casa admin edit page" do
```

Comment:
This message sounds wrong, an unauthenticated request should not access the casa admin page.

Reply:
good catch! now that's the copy&pasta issue 😆